_posts/2019-11-13-variational-convnet/variational_convnet.Rmd

```diff
@@ -0,0 +1,619 @@
+---
+title: "Variational convnets with tfprobability"
+description: > 
+ In a Bayesian neural network, layer weights are distributions, not tensors.
+ Using tfprobability, the R wrapper to TensorFlow Probability, we can build regular Keras models that have probabilistic layers, and thus get uncertainty estimates "for free".
+ In this post, we show how to define, train and obtain predictions from a probabilistic convolutional neural network.
+ 
+author:
+  - name: Sigrid Keydana
+    affiliation: RStudio
+    affiliation_url: https://www.rstudio.com/
+slug: keydana2019variationalconv
+bibliography: bibliography.bib
+date: 11-13-2019
+categories:
+  - Probability and statistics
+  - Time Series
+  - Keras
+output:
+  distill::distill_article:
+    self_contained: false
+preview: images/bbb.png
+---
+
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, eval = FALSE)
+```
+
+
+A bit more than a year ago, in his beautiful [guest post](https://blogs.rstudio.com/tensorflow/posts/2018-07-17-activity-detection/) Nick Strayer showed how to classify a set of everyday activities using smartphone-recorded gyroscope and accelerometer data. Accuracy was very good, but Nick went on to inspect classification results more closely. Were there activities more prone to misclassification than others? And how about those erroneous results: Did the network report them with equal, or less, confidence than those that were correct?
+
+Technically, what we refer to when talking of _confidence_ in that manner is the _score_ obtained for the "winning" class after softmax activation ^[see [Winner takes all: A look at activations and cost functions](https://blogs.rstudio.com/tensorflow/posts/2018-10-11-activations-intro/)]. If that winning score is 0.9, we might say "the network is sure that's a gentoo penguin"; if it's 0.2, we'd instead conclude "to the network, neither option seemed fitting, but cheetah looked best".
+
+This use of "confidence" is convincing, but it has nothing to do with confidence -- or credibility, or prediction, what have you -- intervals. What we'd really like to be able to do is put distributions over the network's weights and make it _Bayesian_. Using _tfprobability_'s variational Keras-compatible layers, this is something we actually can do.
```

Comment:
> confidence -- or credibility, or prediction, what have you -- intervals.

The way this is now makes it sound like they're all different names for the same thing, is that what you're going for? (I genuinely don't know if that's the case in this context).

Reply:
hm, no... of course i don't want to give the impression like to me these are the same...
I'll think about it... on first thought i felt like removing, but "credibility" makes for a good link to "Bayesian"...